### PR TITLE
DHFPROD-1979: Created Json utility classes to promote reuse & changed Step-related …

### PR DESCRIPTION
**Pull request by hao1st**
…classes as an example

https://project.marklogic.com/jira/browse/DHFPROD-1979

@aebadirad @akshaysonvane Please take a look, and let me know...

I ran the StepManagerTest, All were passed.

### CODE REVIEW

File: marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.marklogic.hub.util.json;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class JSONSerializer {
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java]: Can be renamed to JSONStreamWriter

File: marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    private static Logger logger = LoggerFactory.getLogger(JSONSerializer.class);
+
+    final OutputStream out;
+
+    public JSONSerializer(OutputStream out) {
+        this.out = out;
+    }
+
+    public void serialize(Object val) throws IOException {
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java]: Can be renamed to `public void write(Object val)`

File: marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java (hunk 1 above)
Comment (hao1st) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java]: good suggestion.👍

File: marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java (hunk 2 above)
Comment (hao1st) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java]: Yes, match with JSONStreamWriter class.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java (hunk 1 above)
Comment (hao1st) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/util/json/JSONSerializer.java]: The jira # is DHFPROD-1979

**Pull request merged**